pyproject.toml

```diff
@@ -20,32 +17,38 @@ classifiers = [
     "Topic :: Software Development :: Debuggers",
     "Topic :: Internet :: Log Analysis"
 ]
+requires-python = ">=3.11"
+
+dependencies = [
+    "requests>0.2.31",
+    "llama-cpp-python>0.2.56, !=0.2.86",
+    "drain3 >=0.9.11, < 1.0.0",
+    "huggingface-hub>0.23.2",
+    # rawhide has numpy 2, F40 and F41 are still on 1.26
+    # we need to support both versions
+    "numpy>=1.26.0",
+    "python-gitlab >=5.6.0, < 6.0.0",
+]
 
-[tool.poetry.urls]
+[project.urls]
 homepage = "https://github.com/fedora-copr/logdetective"
 issues = "https://github.com/fedora-copr/logdetective/issues"
 
-[tool.poetry.dependencies]
-python = "^3.11"
-requests = ">0.2.31"
-llama-cpp-python = ">0.2.56,!=0.2.86"
-drain3 = "^0.9.11"
-huggingface-hub = ">0.23.2"
-# rawhide has numpy 2, F40 and F41 are still on 1.26
-# we need to support both versions
-numpy = ">=1.26.0"
-python-gitlab = "^5.6.0"
-
-[tool.poetry.group.server.dependencies]
-fastapi = "^0.115.8"
-pydantic = "^2.10.6"
-pyyaml = "^6.0.2"
 
-[build-system]
-requires = ["poetry-core"]
-build-backend = "poetry.core.masonry.api"
 
-[tool.poetry.scripts]
+[project.optional-dependencies]
+server = [
+    "fastapi>=0.115.8, < 1.0.0",
+    "pydantic>=2.10.6, < 3.0.0",
+    "pyyaml>=6.0.2, <7.0.0",
```

Comment:
In order to build for F41, we need to relax the dependency to `"pyyaml>=6.0.1, <7.0.0"`